internal/voucher/voucherpage/vouch_failed.go

```diff
@@ -0,0 +1,43 @@
+package voucherpage
+
+import (
+	"context"
+
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/donor/donordata"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/form"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/lpastore/lpadata"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/notify"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/voucher/voucherdata"
+)
+
+type failVouch func(ctx context.Context, provided *voucherdata.Provided, lpa *lpadata.Lpa) error
+
+func vouchFailed(donorStore DonorStore, notifyClient NotifyClient, appPublicURL string) failVouch {
```

Comment:
the names of these are kind of awkward, but I'm struggling to think of anything better